src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -1001,6 +1016,8 @@ internal sealed partial class BoundDeconstructionAssignmentOperator : BoundExpre
         // TODO: implement IOperation for pattern-matching constructs (https://github.com/dotnet/roslyn/issues/8699)
         protected override OperationKind ExpressionKind => OperationKind.None;
 
+        protected override ImmutableArray<IOperation> Children => ImmutableArray.Create<IOperation>(this.Left, this.Right);
```

Comment:
Note that the `IOperationWithChildren` API is just internal, which only the OperationWalker understands so we can walk down it's children when computing all operations to analyze in a method block. Public clients will still not be able to access any children nodes, we don't need to worry about exposing unwanted nodes here.